cmd/fluentd/lib/fluent/plugin/out_loki.rb

```diff
@@ -259,7 +259,13 @@ def record_to_line(record)
           when :key_value
             formatted_labels = []
             record.each do |k, v|
-              formatted_labels.push(%(#{k}="#{v}"))
+              # Escape double quotes and backslashes by prefixing them with a backslash
+              v = v.to_s.gsub(%r{(["\\])}, '\\\\\1')
+              if v.include?(' ') || v.include?('=')
+                formatted_labels.push(%(#{k}="#{v}"))
+              else
+                formatted_labels.push(%(#{k}=#{v}))
+              end
```

Comment:
I checked some ruby sources for the best way to do this and I came up with this solution instead:

```suggestion
              # This regular expression will validate true if the string doesn't need escaping.
              v = v.dump unless v =~ /\A[\w\.\-\+\%\_\,\:\;\/]*\z/i
              formatted_labels.push(%(#{k}=#{v}))
```

Source: https://github.com/polyfox/moon-logfmt/blob/master/lib/moon-logfmt/utils.rb#L25 and https://ruby-doc.org/core-2.5.1/String.html#method-i-dump